src/v/rpc/simple_protocol.cc

```diff
@@ -159,7 +159,10 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
                     .then([ctx] { ctx->signal_body_parse(); });
               }
 
+              // Grab the method and begin tracking latency.
+              // The context owns the server resources at this point.
               method* m = it->get()->method_from_id(method_id);
+              auto msr = ctx->res.hist().auto_measure();
```

Comment:
It looks like this measures on the same histogram as a few lines down on 168.

Reply:
> It looks like this measures on the same histogram as a few lines down on 168.

@BenPope 

TL;DR -- Correct, however, that histogram is for the latency in other many methods in internal rpc services housed in rpc_compiler.py (previously rpcgen.py).

The histogram on 168 is called for every method within other internal_rpc services such as partition_balancer.
Those services are created in redpanda/application.cc:
```
          proto->register_service<cluster::id_allocator>(...);
          proto->register_service<cluster::tx_gateway>(...);
          proto->register_service<raft::service<cluster::partition_manager, cluster::shard_table>>(...);
          proto->register_service<cluster::service>(...);
          proto->register_service<cluster::metadata_dissemination_handler>(...);
          proto->register_service<cluster::partition_balancer_rpc_handler>(...)
```
and those services are defined/auto-generated in rpc_compiler.py (previously rpcgen.py).

We had a discussion ([here](https://github.com/redpanda-data/redpanda/pull/5411#discussion_r920015427)) about those services in a previous PR. We concluded to not include those methods, and subsequently the hist on 168, in the public endpoint because they would be service-level metrics instead of server-level.


This commit tracks the internal_rpc latency on the server-level and puts it on the public endpoint.

Reply:
On line 229 you can see the hist measurement is recorded into the probe for each service's method.

Reply:
But the `auto_measure` records into the `ctx->res.hist()` at destruction, so it should have already been recorded, and now it's recorded twice?

Reply:
i think ben is referring to 

```
.then_wrapped([ctx, m, l = ctx->res.hist().auto_measure(), rs](
```

rather than the per-method probe down around 229. i have the question about double accounting.

Reply:
The way I'm reading this code is that we're measuring the same thing twice in slightly different ways. I think we should use the same measurement on line 229, but hook the result up in the `net::server_probe`.

Something like:
```
.finally([m, l = std::move(l)]() mutable {
    rs.probe().record_latency(l); // rs needs to be captured somehow 
                                  // and record_latency is a method I invented on net::server_probe
    m->probes.latency_hist().record(std::move(l));
});
```

Reply:
Finally got this working. Surprisingly moving `hdr_hist` into the server_probe led to linker errors with `hdr_init`. Therefore, I kept the hist in the server so the initial problem is fixed. I can work on the linker problem asynchronously.